sdk/devcenter/developer-devcenter-rest/test/public/environmentsTest.spec.ts

```diff
@@ -0,0 +1,125 @@
+  let recorder: Recorder;
+  let client: AzureDevCenterClient;
+  let tenantId: string;
+  let devCenter: string;
+
+  beforeEach(async function (this: Context) {
+    recorder = await createRecorder(this);
+    tenantId = env["DEVCENTER_TENANT_ID"] || "<tenant id>";
+    devCenter = env["DEFAULT_DEVCENTER_NAME"] || "sdk-default-devcenter";
+    client = createRecordedClient(recorder, tenantId, devCenter, {
+      allowInsecureConnection: false,
+    });
+  });
+
+  afterEach(async function () {
+    await recorder.stop();
+  });
+
+  it("Create environment", async function () {
+    // Build client and fetch required parameters
+    const projectName = env["DEFAULT_PROJECT_NAME"] || "sdk-default-project";
+    const catalogName = env["DEFAULT_CATALOG_NAME"] || "sdk-default-catalog";
+    const catalogItemName = env["DEFAULT_CATALOG_ITEM_NAME"] || "Empty";
+    const environmentTypeName =
+      env["DEFAULT_ENVIRONMENT_TYPE_NAME"] || "sdk-default-environment-type";
+    const environmentName = "SdkTest-Environment";
+    const userId = "me";
+
+    console.log(
+      `Running test for ${tenantId} -- ${devCenter} -- ${projectName} -- ${catalogName} -- ${catalogItemName} -- ${environmentTypeName} -- ${environmentName}`
+    );
+
+    const environmentsCreateParameters: EnvironmentsCreateOrUpdateEnvironmentParameters = {
+      contentType: "application/json",
+      body: {
+        catalogItemName: catalogItemName,
+        environmentType: environmentTypeName,
+        catalogName: catalogName,
+      },
+    };
+
+    // Provision a dev box
+    const environmentCreateResponse = await client
+      .path(
+        "/projects/{projectName}/users/{userId}/environments/{environmentName}",
+        projectName,
+        userId,
+        environmentName
+      )
+      .put(environmentsCreateParameters);
+    console.log("Sent create");
+    assert.equal(
+      environmentCreateResponse.status,
+      "201",
+      "Environment creation should return 201 created."
+    );
+    if (isUnexpected(environmentCreateResponse)) {
+      throw new Error(
+        `Creation failed with message ${environmentCreateResponse.body?.error.message}`
+      );
+    }
+
+    const environmentCreatePoller = getLongRunningPoller(client, environmentCreateResponse);
+    const environmentCreateResult = await environmentCreatePoller.pollUntilDone();
+
+    assert.equal(environmentCreateResult.body.provisioningState, "Succeeded");
+    console.log(
+      `Provisioned environment with state ${environmentCreateResult.body.provisioningState}.`
+    );
+
+    // Get the deployment outputs
+    const artifactListResult = await client
+      .path(
+        "/projects/{projectName}/users/{userId}/environments/{environmentName}/artifacts",
+        projectName,
+        userId,
+        environmentName
+      )
+      .get();
+    assert.equal(artifactListResult.status, "200", "Artifact listing should return 200 OK.");
```

Comment:
I would suggest you to do this assertion after the `isUnexpected` and property check for the returned result